src/main/java/com/spotify/dns/DnsSrvResolvers.java

```diff
@@ -78,6 +82,10 @@ public DnsSrvResolver build() {
 
       LookupFactory lookupFactory = new SimpleLookupFactory();
 
+      if (!useLookupCache) {
```

Comment:
A test validating the conflict handling would also solve the code coverage regression.